tests/test_m2m.py

```diff
@@ -0,0 +1,159 @@
+
+    def testMapQualiiedBasicWithFunction(self):
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.map("basic", "title", "test_title", qualifier="officialtitle", function=(lambda x: x.upper()))
+
+        self.assertEquals(self.record.__str__(), '<?xml version="1.0" encoding="UTF-8"?>\n<metadata>\n  <title qualifier="officialtitle">TEST_TITLE</title>\n  <meta qualifier="metadataCreator">mphillips</meta>\n</metadata>\n')
+
+    def testMapUnqualifiedAgentNoOptions(self):
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.map("agent", "creator", "Phillips, Mark")
+
+        self.assertEquals(self.record.__str__(), '<?xml version="1.0" encoding="UTF-8"?>\n<metadata>\n  <creator>\n    <name>Phillips, Mark</name>\n  </creator>\n  <meta qualifier="metadataCreator">mphillips</meta>\n</metadata>\n')
+
+    def testMapQualifiedAgentNoOptions(self):
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.map("agent", "creator", "Phillips, Mark", qualifier="aut")
+
+        self.assertEquals(self.record.__str__(), '<?xml version="1.0" encoding="UTF-8"?>\n<metadata>\n  <creator qualifier="aut">\n    <name>Phillips, Mark</name>\n  </creator>\n  <meta qualifier="metadataCreator">mphillips</meta>\n</metadata>\n')
+
+    def testMapQualifiedAgentInfoOnly(self):
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.map("agent", "creator", "Phillips, Mark", 
+                        qualifier="aut", info="First Publication")
+
+        self.assertEquals(self.record.__str__(), '<?xml version="1.0" encoding="UTF-8"?>\n<metadata>\n  <creator qualifier="aut">\n    <info>First Publication</info>\n    <name>Phillips, Mark</name>\n  </creator>\n  <meta qualifier="metadataCreator">mphillips</meta>\n</metadata>\n')
+
+    def testMapQualifiedAgentInfoAndAgentType(self):
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.map("agent", "creator", "Phillips, Mark", 
+                        qualifier="aut",
+                        info="First Publication",
+                        agent_type="per")
+
+        self.assertEquals(self.record.__str__(), '<?xml version="1.0" encoding="UTF-8"?>\n<metadata>\n  <creator qualifier="aut">\n    <info>First Publication</info>\n    <type>per</type>\n    <name>Phillips, Mark</name>\n  </creator>\n  <meta qualifier="metadataCreator">mphillips</meta>\n</metadata>\n')
+
+    def testMapQualifiedAgentWithLocation(self):
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.map("agent", "publisher", "UNT Libraries", 
+                        location="Denton, Texas")
+
+        self.assertEquals(self.record.__str__(), '<?xml version="1.0" encoding="UTF-8"?>\n<metadata>\n  <publisher>\n    <name>UNT Libraries</name>\n    <location>Denton, Texas</location>\n  </publisher>\n  <meta qualifier="metadataCreator">mphillips</meta>\n</metadata>\n')
```

Comment:
It is probably a better idea to use the `str()` built-in in this assertion and the similar assertions in the preceding tests.

```python 
self.assertEquals(str(self.record), '<?xml .... >')